### ISSUE TITLE
warn about pointless classes

### ISSUE THREAD

**Issue by chdsbd** [sbdchd/flake8-pie#46]
If a class has no state and only classmethods and/or static methods, it doesn't need to be a class. The methods can be normal functions.

Here's a complete nonsense example that covers the use case.

```python
from typing import NamedTuple


class UserManager:
    """
    some class
    """

    class User(NamedTuple):
        name: str

    @classmethod
    def update_user(cls, user: User) -> None:
        ...

    @staticmethod
    def sync_users() -> None:
        ...
```

This should be:

```python
from typing import NamedTuple

class User(NamedTuple):
    name: str


def update_user(user: User) -> None:
    ...


def sync_users() -> None:
    ...
```

**Comment by sbdchd** [comment]
This is a made up example, but wondering if this could be something that could trigger a false positive


```python
from typing import Protocol
class User:
    ...
    
UserId = str

class DBApi:
     @classmethod
     def create_user(cls, user: User) -> None: ...
     @classmethod
     def update_user(cls, user: User) -> None: ...
     @staticmethod
     def delete_user(user_id: UserId) -> None: ...
     
     
class Api(Protocol):
     @classmethod
     def create_user(cls, user: User) -> None: ...
     @classmethod
     def update_user(cls, user: User) -> None: ...
     @staticmethod
     def delete_user(user_id: UserId) -> None: ...


def send_message(api: Api) -> bool: ...


send_message(DBApi())
```
